robot/tests/test_general.robot

```diff
@@ -0,0 +1,108 @@
+*** Settings ***
+Documentation		This suite will verifiy basic ipmitool related test
+...                     cases. It includes test case for testing power on/off,
+...                     reset, sol etc.
+
+Resource                ../lib/ipmi_client.robot
+Resource                ../lib/utils.robot
+Resource                ../lib/connection_client.robot
+
+Suite Setup             Open Connection And Log In
+Suite Teardown          Close All Connections
+
+Test Setup              Pre Test Execution Setup
+Test Teardown           Expected Result verification
+
+
+*** Variables ***
+
+
+*** Test Cases ***                                
+
+Restart_using_NMI
+    [Documentation]   Restart using NMI
+
+    Log To Console       \n Triggering NMI
+    Run IPMI Standard Command    power diag
+    Sleep   60sec
+
+    Wait for OS
+
+
+Multiple times Activate and Deactivate SOL
+    [Documentation]   Continous IPMItool command to BMC
+    : FOR    ${count}    IN RANGE    0    15
+    \    Log To Console  \n [ *** IPMItool command count: ${count} *** ]
+    \    Run IPMI Standard Command    sol deactivate
+    \
+    \    ${resp}=    Run IPMI Standard Command    sol activate &
+    \    Log To Console    ${resp}
+    \    Should Contain    ${resp}    SOL Session operational
+
+
+Continous IPMItool command to BMC
+    [Documentation]   Continous IPMItool command to BMC
+    : FOR    ${count}    IN RANGE    0    15
+    \    Log To Console  \n [ *** IPMItool command count: ${count} *** ]
+    \    ${resp}=    Run IPMI Standard Command    sol info
+    \    Should Contain    ${resp}    Set in progress
+
+
+Multiple Power OFF and ON
+    [Documentation]   Multiple iteration of power off and on
+    : FOR    ${count}    IN RANGE    0    15
```

Comment:
Where does 15 come from?
How long does this take? Quick math on some machines says this could take over an hour on some machines, which puts it out of the question for a quick regression test. How do you propose to split normal regression tests from longer running tests?